Doc/library/stdtypes.rst

```diff
@@ -5162,6 +5162,15 @@ enables cleaner type hinting syntax compared to :data:`typing.Union`.
       def square(number: int | float) -> int | float:
           return number ** 2
 
+   .. note::
+
+      The object on both sides of the ``|`` operand must be an object that
```

Comment:
That's not exactly true:

```
>>> int | 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'type' and 'int'
>>> (1).__or__
<method-wrapper '__or__' of int object at 0x104596f90>
```

Instead, only the following objects are accepted (this is hardcoded in the `is_unionable` function in `unionobject.c`):
- Instances of `type`
- Instances of `types.GenericAlias`
- Instances of `typing.TypeAliasType`
- Instances of `types.UnionType`
- `None`

Reply:
> Instead, only the following objects are accepted (this is hardcoded in the `is_unionable` function in `unionobject.c`):

Yeah. But then, of course, other types also support PEP-604 methods (e.g. instances of TypeVar), they just do it by defining `__or__` and `__ror__` methods that return typing.Union.

It all gets a bit complicated, so I think I'll try to take some of the detail out of this note and just say "strings aren't supported".